backend/resources/otpResource.js

```diff
@@ -0,0 +1,88 @@
+import express from 'express'
+import nodemailer  from 'nodemailer'
+import dotenv from 'dotenv'
+import * as errorConstants from '../constants/errorMessages.js';
+import * as mailOption from '../constants/mailOption.js';
+
+const otpRouter = express.Router();
+
+dotenv.config();
+
+// Nodemailer configuration using Ethereal
+async function getTestAccount() {
+  return nodemailer.createTestAccount();
+}
+
+// Generate a random 4-digit OTP
+function generateOTP() {
+  return Math.floor(1000 + Math.random() * 9000).toString();
+}
+
+// Nodemailer configuration
+const transporter = nodemailer.createTransport({
+  // Create an ethereal account and replace it in .env file 
+  host: process.env.SMTP_HOST,
+  port: process.env.SMTP_PORT,
+  auth: {
+    user: process.env.SMTP_USER,
+    pass: process.env.SMTP_PASS,
+  },
+});
+
+// Store generated OTPs temporarily (for demo purposes)
+const otpStorage = new Map();
```

Comment:
consider storing otp's for a user in its mongodb document (variables states are generally lost in serverless architectures.)